bentoml/_internal/bento/build_config.py

```diff
@@ -369,29 +385,87 @@ def with_defaults(self) -> "CondaOptions":
         return attr.evolve(self, **defaults)
 
 
+def _conda_options_structure_hook(d: t.Any, _: t.Type[CondaOptions]) -> CondaOptions:
+    pip_list: t.List[str]
+    if "pip" in d and d["pip"] is not None:
+        pip_list = d["pip"]
+    else:
+        pip_list = []
+    if "dependencies" in d and d["dependencies"] is not None:
+        deps: DependencyType = d["dependencies"]
+        if not isinstance(deps, list):
+            raise InvalidArgument(
+                f"dependencies has to be type list, got type {type(d['dependencies'])}"
+            )
+        for dep in deps:
+            if isinstance(dep, dict):
+                assert "pip" in dep and len(dep.keys()) == 1
+                pip_from_deps = dep.pop("pip")
+                pip_list.extend(pip_from_deps)
+        deps = list(filter(lambda x: not isinstance(x, dict), deps))
+        d["dependencies"] = deps
+    d["pip"] = None if not pip_list else pip_list
+
+    return CondaOptions(**d)
+
+
+bentoml_cattr.register_structure_hook(CondaOptions, _conda_options_structure_hook)
```

Comment:
this is probably not necessary, I think we can just store the CondaOptions the way the user specified it.